### ISSUE TITLE
Grouping of uncertain values

### ISSUE THREAD

**Issue by kahaaga** [kahaaga/UncertainData.jl#36]
Some times one might have samples whose age distributions are not identical, even though there is good reason to believe the age distributions represent the same event. 

An example is having C14 dates on carbonate shells found within the same debris flow (a single, short-lasting event) in a sediment core) with non-identical age distributions. 

In these cases, the user should be able to group the distributions when constructing an uncertain dataset. 

The easiest way to do this is probably to provide a vector of uncertain values that are combined using resampling and a kernel density estimate to the resulting sample. 

The syntax for creating an uncertain dataset with grouped values could be 

```julia 
# Create some age distributions that are known represent different events 
distinct_event_distributions = [UncertainValue(Normal, rand(), rand())]

# We also have a few events that are represented by multiple age distributions 
event1_distributions = [UncertainValue(Normal, 1, 0.3), UncertainValue(Normal, 0.8, 0.4)]
event2_distributions = [UncertainValue(Normal, 4, 0.5), UncertainValue(Normal, 3.7, 0.3)]

# Merge distributions representing the same events. Defaults to a kernel density estimate 
# of a resampling of all the members of the groups. Could be specialized for certain types 
# of uncertain values.
group1 = group(event_1_distributions)
group2 = group(event_2_distributions)

# Specifying the number of samples when grouping (may not be necessary for all distributions)
group1 = group(event_1_distributions)
group2 = group(event_2_distributions)


# Construct an uncertain dataset. 
udata = UncertainDataset([distinct_event_distributions; group1; group2])
```

**Comment by kahaaga** [comment]
It's probably better to name this function `merge_overlapping` or something along those lines to not conflict with other packages. Alternatively, might extend `Base.merge`.

**Comment by kahaaga** [comment]
Addressed in #53.

**Comment by kahaaga** [comment]
Added in release v0.2.1.
